frontend/openchat-client/src/openchat.ts

```diff
@@ -3374,21 +3374,35 @@ export class OpenChat {
                     chatId: serverChat.id,
                     chatLastUpdated: lastUpdated,
                 }).catch(CommonResponses.failure);
-                if ("members" in resp && resp.timestamp > lastUpdated) {
+                if ("members" in resp) {
+                    if (!chatIdentifiersEqual(serverChat.id, selectedChatIdStore.value)) {
+                        console.warn(
+                            "Attempting to set chat details on the wrong chat - probably a stale response",
+                            serverChat.id,
+                            selectedChatIdStore.value,
+                        );
+                        return;
+                    }
+                    if (selectedServerChatStore.value?.chatId === serverChat.id && resp.timestamp <= lastUpdated) {
```

Comment:
I don't get why we have this check on chat Id when we have already done the check above

Reply:
In fact I don't think I get this change at all

Reply:
No wait - I _do_ get it. But it should be using `chatIdentifiersEqual` should it not?

Reply:
The check above is on `selectedChatIdStore`, but then the store we update is the `selectedServerChatStore`, and this is the only place it is updated, so when you change chat we always need to update it, but after that, we only update it if there have actually been updates.